src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -117,11 +117,13 @@ UCS_PROFILE_FUNC(ucs_status_t, uct_cuda_copy_mem_reg,
 
     log_level = (flags & UCT_MD_MEM_FLAG_HIDE_ERRORS) ? UCS_LOG_LEVEL_DEBUG :
                 UCS_LOG_LEVEL_ERROR;
-    status    = UCT_CUDA_FUNC(cudaHostRegister(address, length,
-                                               cudaHostRegisterPortable),
-                              log_level);
-    if (status != UCS_OK) {
-        return status;
+    result = cuMemHostRegister(address, length, CU_MEMHOSTREGISTER_PORTABLE);
```

Comment:
is there a reason to change `cudaHostRegister` to `cuMemHostRegister`?
Do we need to use [cuMemHostUnregister()](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__MEM.html#group__CUDA__MEM_1g63f450c8125359be87b7623b1c0b2a14) in `uct_cuda_copy_mem_dereg` then?

Reply:
@brminich No particular reason other than trying to rely on only driver API. Will replace unregister call with driver API as well.